lib/functions.php

```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
+ * @param $transmitter: the transmitter of the email
+ * @param $recipient: the recipient of the email
+ * @param $message is the message of the email
+ * @return true if the mail send.
+ */
+function jirafeau_send_mail($transmitter, $recipient, $message, $link, $email_subject, $filename, $expireDate, $password)
+{
+    $message = '<html>
+    <style>
+        #h1 {
+            color:#663d1c;
+        }
+        #body {
+            justify-content:center;
+            display: flex;
+        }
+        #strong {
+            color: #663d1c;
+        }
+        #expireFile {
+            font-size: 14px;
+            margin-top: 0px;
+        }
+        .h3 {
+            margin: 0px;
+        }
+        .block {
+            display: flex;
+            flex-direction: column;
+            align-items: center;
+            margin-top:20px;
+            margin-bottom:20px;
+        }
+        #text {
+            margin: 30px;
+        }
+        #button {
+            text-transform: uppercase;
+            outline: 2px;
+            background: #663d1c;
+            width: 35%;
+            padding: 15px;
+            color: #FFFFFF;
+            font-size: 14px;
+            cursor: pointer;
+            border-radius: 20px;
+        }
+        #footer {
+            font-size: 12px;
+            display: flex;
+            flex-direction:column;
+            align-items: center;
+        }
+        #mail {
+            background-color:#f4f4f4;
+            max-width: 60%;
+            display:flex;
+            flex-direction: column;
+            align-items:center;
+        }
+    </style>
+    <head>
+        <title>Jirafeau</title>
+    </head>
+    <body id="body">
+        <div id="mail">
+            <h1 id="h1">Jirafeau</h1>
+            <img src="https://framalibre.org/sites/default/files/leslogos/JirafeauLogo.jpeg">
+            <div class="block">
+                <h2>
+                    <strong id="strong">' . $transmitter . '</strong> sent you a file
+                </h2>
+                <p id="expireFile">Expires on ' . $expireDate . '</p>
+            </div>
+            <div id="text">
+                <p>' . $message .'</p>
+            </div>
+            <button id="button" href=' . $link .'>Get your file</button>
+            <div class="block">
+                <h3 class="h3">Dowload link:</h3>
+                <p>' . $link . '</p>
+                <h3 class="h3">Files: ' . $filename .'</h3>
+                <h4>Password:' . $password . '</h4>
+            </div>
+        </div>
+    </body>
+    </html>';
+    $headers[] = 'From: '.$transmitter;
+    $headers[] = "Reply-to:" . $transmitter;
+    $headers[] = 'To: '.$recipient;
+    $headers[] = "Importance: Normal";
+    $headers[] = "MIME-Version: 1.0";
+    $headers[] = 'Content-Type: text/html; charset="utf-8"';
+    $headers[] = 'Content-Tranfert-Encoding: 8bit';
+    return mail($recipient, $email_subject, $message, implode("\r\n", $headers));
+}
```

Comment:
I don't understand how to manage an array of recipient. Do you send one mail per recipient, or one for every recipients (in "To" or in "CC" or in "CCI").

Reply:
the recipients are retrieved using 
`let NbrRecipients = document.getElementsByName('recipient'); ` , and added to the recipient array in a for :
```
for (element of NbrRecipients) {
        recipient.push (element.value);
    }
```
Then the array is sent with the xhr request, and it is received in script.php as a comma parser string (that also surprised me) sending mail is done for each recipient, they are all in the box "To"
![Screenshot_20210212_153411](https://user-images.githubusercontent.com/71255044/107781505-42292e00-6d48-11eb-8bcd-c304a0013a9e.png)
![Screenshot_20210212_153z411](https://user-images.githubusercontent.com/71255044/107781506-42c1c480-6d48-11eb-97f2-fdc19b9e912d.png)
![Screenshot_2021-02-12 regtferdbhgtfe - guillaumerand gmail com - Gmail](https://user-images.githubusercontent.com/71255044/107781953-c8457480-6d48-11eb-8f5c-9b9c03931d4a.png)